runtime/sema/type.go

```diff
@@ -6019,6 +6024,9 @@ func (t *ReferenceType) String() string {
 	if t.Authorization != UnauthorizedAccess {
 		authorization = t.Authorization.String()
 	}
+	if _, isMapping := t.Authorization.(*EntitlementMapAccess); isMapping {
+		authorization = "mapping " + authorization
```

Comment:
Maybe this should be part of the authorization `String`/`QualifiedString` functions?

Reply:
Those are used to compute the TypeIDs for `EntitlementMapAccess`es, and we want those not to include the "mapping" prefix since they are used for map lookups/indexing.